test/irb/test_ruby_lex.rb

```diff
@@ -229,9 +229,9 @@ def test_heredoc_with_embexpr
       input_with_prompt = [
         PromptRow.new('001:0:":* ', %q(<<A+%W[#{<<B)),
         PromptRow.new('002:0:":* ', %q(#{<<C+%W[)),
-        PromptRow.new('003:0:":* ', %q()),
-        PromptRow.new('004:0:":* ', %q(C)),
```

Comment:
If I understand it correctly, this prompt should be `]` if Ripper works as expected?

Reply:
Yes. after `C`, heredoc `<<C` is closed and its inside `%W[` so the prompt should be `]`.

Reply:
Thanks for explaining 👍 
Can we add a todo comment to add the this case back as well after the patch is merged?

Reply:
If we change it back, I think test passes in ruby head but fails in ruby 2.7~3.2 (unless it's backported) so I want to let this test keep using `%q(a)`.

The reason why I add `%q()` here is to check the prompt between `%W[`, `heredoc_close`, `]`, so changing `%q()` to `%q(a)` still play a role. It does not have to be an empty string.
Test seems to properly fails before this fix #390.

Reply:
Oh I thought that patch will be backported. But perhaps not to 2.7 anyway 🤔 . Yeah so ignore my comment then 😄